Sources/NIOKit/NIOKitTestCase.swift

```diff
@@ -0,0 +1,35 @@
+import NIO
+import XCTest
+
+protocol NIOKitTestCaseProtocol: class {
+    var group: EventLoopGroup! { get set }
+}
+
+extension NIOKitTestCaseProtocol {
+    public func setupGroup() {
+        self.group = MultiThreadedEventLoopGroup(numberOfThreads: 1)
+    }
+    
+    public func teardownGroup() {
+        XCTAssertNoThrow(try self.group.syncShutdownGracefully())
+        self.group = nil
+    }
+}
+
+open class NIOKitTestCase: XCTestCase, NIOKitTestCaseProtocol {
```

Comment:
This should be `public` so that we can use it without `@testable`. There are a few situations where this is useful:
- `swift test -c release` (doesn't work with @testable on linux)
- Tests which must only use public API

Reply:
open supersedes public right? So anything `open` is also `public`? Or did I get my terminology messed up

Reply:
Oh duh, yes you're right. My mind just inserted `final` there because I'm so used to that haha.